crates/libcontainer/src/container/builder_impl.rs

```diff
@@ -121,7 +122,8 @@ impl<'a> ContainerBuilderImpl<'a> {
             cgroup_manager: cmanager,
         };
 
-        let init_pid = process::container_main_process::container_main_process(&container_args)?;
+        let (intermediate, init_pid) =
```

Comment:
Maybe I missed this, but can you add a bit of comment on the need to return the `intermediate` pid as well?

Reply:
can you take a look at this https://github.com/utam0k/youki/pull/115/files#diff-4c3bee02d644d36778359cdb85ba23c194655461d95e34c298659d4633c2459eR101-R106